src/Sentry/Internal/DebugStackTrace.cs

```diff
@@ -382,77 +382,116 @@ private static void DemangleLambdaReturnType(SentryStackFrame frame)
     private int? AddDebugImage(Module module)
     {
         var id = module.ModuleVersionId;
-
         if (_debugImageIndexByModule.TryGetValue(id, out var idx))
         {
             return idx;
         }
 
+        var debugImage = GetDebugImage(module, _options);
+        if (debugImage == null)
+        {
+            // don't try to resolve again
+            _debugImageIndexByModule.Add(id, DebugImageMissing);
+            return null;
+        }
+
+        idx = DebugImages.Count;
+        DebugImages.Add(debugImage);
+        _debugImageIndexByModule.Add(id, idx);
+
+        return idx;
+    }
+
+    internal static DebugImage? GetDebugImage(Module module, SentryOptions options)
```

Comment:
Note this is also just refactoring.  The `AddDebugImage` method now is entirely responsible for managing the cache, while `GetDebugImage` can focus on the hard parts.  (We can later add some tests on `GetDebugImage`.)